tools/db_bench_tool.cc

```diff
@@ -789,6 +789,27 @@ DEFINE_int32(min_level_to_compress, -1, "If non-negative, compression starts"
              " not compressed. Otherwise, apply compression_type to "
              "all levels.");
 
+typedef enum rocksdb::BlockBasedTableOptions::DataBlockIndexType
+    DataBlockIndexType;
+
+static DataBlockIndexType StringToDataBlockIndexType(const char* ctype) {
+  assert(ctype);
+
+  if (!strcasecmp(ctype, "binary"))
+    return rocksdb::BlockBasedTableOptions::kDataBlockBinarySearch;
+  else if (!strcasecmp(ctype, "binary_and_hash"))
+    return rocksdb::BlockBasedTableOptions::kDataBlockBinaryAndHash;
+
+  fprintf(stdout, "Cannot parse data block index type '%s'\n", ctype);
+
+  // return default value
+  return rocksdb::BlockBasedTableOptions::kDataBlockBinarySearch;
+}
+
+DEFINE_string(data_block_index_type, "binary", "Index type for data blocks");
```

Comment:
Since we have only two values for index type at the moment, it seems like defining this as a bool might be simpler. Take a look at how "use_hash_search" option is defined. If you take that approach you can avoid the string comparison and parsing too.